gmso/abc/gmso_base.py

```diff
@@ -54,6 +56,44 @@ def __init_subclass__(cls, **kwargs):
                 )
         apply_docs(cls, map_names=True, silent=False)
 
+    @classmethod
+    def parse_obj(cls: Type["Model"], obj: Any) -> "Model":
+        dict_to_unyt(obj)
+        return super(GMSOBase, cls).parse_obj(obj)
```

Comment:
In terms of common errors I see with handling these json files, if something gets modified then the file won't be able to be read back into the proper gmso core class. For example, if I change a unyt from `"unit": "C"` to `"unit": "Col"`, then unyt won't be able to read that back in. However, the current error message that is thrown in this case is a bit cryptic for a normal user:
"""python
> from gmso import Dihedral
> with open('test_c_atom.json', 'r') as json_file:
>     atom_json_raw = Dihedral.parse_raw(json_file.read())
"""
ValidationError                           Traceback (most recent call last)
<ipython-input-29-36431a19f079> in <module>
      2 from gmso import Dihedral
      3 with open('test_c_atom.json', 'r') as json_file:
----> 4     atom_json_raw = Dihedral.parse_raw(json_file.read())

~/miniconda3/envs/gmso-dev/lib/python3.7/site-packages/pydantic/main.cpython-37m-darwin.so in pydantic.main.BaseModel.parse_raw()

~/Documents/Vanderbilt/Research/gmso/gmso/abc/gmso_base.py in parse_obj(cls, obj)
     60     def parse_obj(cls: Type["Model"], obj: Any) -> "Model":
     61         dict_to_unyt(obj)
---> 62         return super(GMSOBase, cls).parse_obj(obj)
     63 
     64     def dict(self, **kwargs) -> "DictStrAny":

~/miniconda3/envs/gmso-dev/lib/python3.7/site-packages/pydantic/main.cpython-37m-darwin.so in pydantic.main.BaseModel.parse_obj()

~/miniconda3/envs/gmso-dev/lib/python3.7/site-packages/pydantic/main.cpython-37m-darwin.so in pydantic.main.BaseModel.__init__()

ValidationError: 1 validation error for Dihedral
__root__
  A non-site object provided to be a connection member (type=type_error)
> 
If it's simple enough, it could be worth it to write a more descriptive error message. That way users will easily know there's just a typo somewhere within the json file somehow.

Reply:
A more descriptive error message has been added in 381df27.